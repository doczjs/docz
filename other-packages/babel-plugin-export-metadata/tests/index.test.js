const { transformSync } = require('@babel/core')
const fs = require('fs')
const path = require('path')
const plugin = require('../src')

const exportNamedFixture = path.resolve(
  './tests/fixtures/export-named/index.js'
)

const exportDefaultFixtures = {
  withArrExpression: path.resolve(
    './tests/fixtures/export-default/with-arr-expression.js'
  ),
  withClassDeclaration: path.resolve(
    './tests/fixtures/export-default/with-class-declaration.js'
  ),
  withFuncDeclaration: path.resolve(
    './tests/fixtures/export-default/with-func-declaration.js'
  ),
  withIdentifier: path.resolve(
    './tests/fixtures/export-default/with-identifier.js'
  ),
  withObjExpression: path.resolve(
    './tests/fixtures/export-default/with-obj-expression.js'
  ),
  withCallExpression: path.resolve(
    './tests/fixtures/export-default/with-call-expression.js'
  ),
}

const reExportsFixtures = {
  reExportName: path.resolve(
    path.resolve('./tests/fixtures/re-export/re-export.js')
  ),
  reExportDefault: path.resolve(
    path.resolve('./tests/fixtures/re-export/re-export-default.js')
  ),
  reExportDefaultToName: path.resolve(
    path.resolve('./tests/fixtures/re-export/re-export-rename1.js')
  ),
  reExportNameToDefault: path.resolve(
    path.resolve('./tests/fixtures/re-export/re-export-rename2.js')
  ),
  reExportNameToName: path.resolve(
    path.resolve('./tests/fixtures/re-export/re-export-rename3.js')
  ),
}

const getCodeFromFilePath = paths =>
  Object.keys(paths).reduce(
    (acc, key) => ({
      ...acc,
      [key]: fs.readFileSync(paths[key]).toString(),
    }),
    {}
  )

const exportDefaultCode = getCodeFromFilePath(exportDefaultFixtures)
const exportNamedCode = fs.readFileSync(exportNamedFixture).toString()
const reExportCode = getCodeFromFilePath(reExportsFixtures)

describe('export-metadata', () => {
  describe('re-export', () => {
    it('re-export default', () => {
      const result = transformSync(exportDefaultFixtures.reExportName, {
        plugins: [plugin],
        filename: exportNamedFixture.reExportName,
      })

      expect(result.code).toMatchSnapshot()
    })
  })

  describe('export default', () => {
    it('works with Array expression', () => {
      const result = transformSync(exportDefaultCode.withArrExpression, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withArrExpression,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('works with Object expression', () => {
      const result = transformSync(exportDefaultCode.withObjExpression, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withObjExpression,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('works with Class declaration', () => {
      const result = transformSync(exportDefaultCode.withClassDeclaration, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withClassDeclaration,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('works with Function declaration', () => {
      const result = transformSync(exportDefaultCode.withFuncDeclaration, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withFuncDeclaration,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('works with Identifier', () => {
      const result = transformSync(exportDefaultCode.withIdentifier, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withIdentifier,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('works with Call expression', () => {
      const result = transformSync(exportDefaultCode.withCallExpression, {
        plugins: [plugin],
        filename: exportDefaultFixtures.withIdentifier,
      })

      expect(result.code).toMatchSnapshot()
    })
  })

  describe('export named', () => {
    it('works', () => {
      const result = transformSync(exportNamedCode, {
        plugins: [plugin],
        filename: exportNamedFixture,
      })

      expect(result.code).toMatchSnapshot()
    })
  })

  describe('re-export', () => {
    it('re export name', () => {
      const result = transformSync(reExportCode.reExportName, {
        plugins: [plugin],
        filename: reExportsFixtures.reExportName,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('re export default', () => {
      const result = transformSync(reExportCode.reExportDefault, {
        plugins: [plugin],
        filename: reExportsFixtures.reExportDefault,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('re export name to default', () => {
      const result = transformSync(reExportCode.reExportNameToDefault, {
        plugins: [plugin],
        filename: reExportsFixtures.reExportNameToDefault,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('re export default to name', () => {
      const result = transformSync(reExportCode.reExportDefaultToName, {
        plugins: [plugin],
        filename: reExportsFixtures.reExportDefaultToName,
      })

      expect(result.code).toMatchSnapshot()
    })

    it('re export name to name', () => {
      const result = transformSync(reExportCode.reExportNameToName, {
        plugins: [plugin],
        filename: reExportsFixtures.reExportNameToName,
      })

      expect(result.code).toMatchSnapshot()
    })
  })
})
