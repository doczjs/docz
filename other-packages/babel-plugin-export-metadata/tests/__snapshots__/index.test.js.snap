// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-metadata export default works with Array expression 1`] = `
"/* ExportDefaultDeclaration with Array expression */
let foo5 = 5;

let bar5 = () => {};

export default [foo5, bar5];

if (typeof bar5 !== 'undefined' && bar5 && bar5 === Object(bar5) && Object.isExtensible(bar5) && !bar5.hasOwnProperty('__filemeta')) {
  Object.defineProperty(bar5, '__filemeta', {
    configurable: true,
    value: {
      name: \\"bar5\\",
      filename: \\"tests/fixtures/export-default/with-arr-expression.js\\"
    }
  });
}

if (typeof foo5 !== 'undefined' && foo5 && foo5 === Object(foo5) && Object.isExtensible(foo5) && !foo5.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo5, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo5\\",
      filename: \\"tests/fixtures/export-default/with-arr-expression.js\\"
    }
  });
}"
`;

exports[`export-metadata export default works with Call expression 1`] = `
"/* ExportDefaultDeclaration with Call expression */
const foo = v => v;

const __DOCZ_DUMMY_EXPORT_DEFAULT = foo(5);

export default __DOCZ_DUMMY_EXPORT_DEFAULT;

if (typeof __DOCZ_DUMMY_EXPORT_DEFAULT !== 'undefined' && __DOCZ_DUMMY_EXPORT_DEFAULT && __DOCZ_DUMMY_EXPORT_DEFAULT === Object(__DOCZ_DUMMY_EXPORT_DEFAULT) && Object.isExtensible(__DOCZ_DUMMY_EXPORT_DEFAULT) && !__DOCZ_DUMMY_EXPORT_DEFAULT.hasOwnProperty('__filemeta')) {
  Object.defineProperty(__DOCZ_DUMMY_EXPORT_DEFAULT, '__filemeta', {
    configurable: true,
    value: {
      name: \\"__DOCZ_DUMMY_EXPORT_DEFAULT\\",
      filename: \\"tests/fixtures/export-default/with-identifier.js\\"
    }
  });
}"
`;

exports[`export-metadata export default works with Class declaration 1`] = `
"/* ExportDefaultDeclaration with Class declaration */
export default class Bar6 {}

if (typeof Bar6 !== 'undefined' && Bar6 && Bar6 === Object(Bar6) && Object.isExtensible(Bar6) && !Bar6.hasOwnProperty('__filemeta')) {
  Object.defineProperty(Bar6, '__filemeta', {
    configurable: true,
    value: {
      name: \\"Bar6\\",
      filename: \\"tests/fixtures/export-default/with-class-declaration.js\\"
    }
  });
}"
`;

exports[`export-metadata export default works with Function declaration 1`] = `
"/* ExportDefaultDeclaration with Function declaration */
export default function foo6() {}

if (typeof foo6 !== 'undefined' && foo6 && foo6 === Object(foo6) && Object.isExtensible(foo6) && !foo6.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo6, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo6\\",
      filename: \\"tests/fixtures/export-default/with-func-declaration.js\\"
    }
  });
}"
`;

exports[`export-metadata export default works with Identifier 1`] = `
"/* ExportDefaultDeclaration with Identifier */
let foo3 = 5;
export default foo3;

if (typeof foo3 !== 'undefined' && foo3 && foo3 === Object(foo3) && Object.isExtensible(foo3) && !foo3.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo3, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo3\\",
      filename: \\"tests/fixtures/export-default/with-identifier.js\\"
    }
  });
}"
`;

exports[`export-metadata export default works with Object expression 1`] = `
"/* ExportDefaultDeclaration with Object expression */
export default {
  foo4: 5,
  bar4: () => {}
};

if (typeof bar4 !== 'undefined' && bar4 && bar4 === Object(bar4) && Object.isExtensible(bar4) && !bar4.hasOwnProperty('__filemeta')) {
  Object.defineProperty(bar4, '__filemeta', {
    configurable: true,
    value: {
      name: \\"bar4\\",
      filename: \\"tests/fixtures/export-default/with-obj-expression.js\\"
    }
  });
}

if (typeof foo4 !== 'undefined' && foo4 && foo4 === Object(foo4) && Object.isExtensible(foo4) && !foo4.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo4, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo4\\",
      filename: \\"tests/fixtures/export-default/with-obj-expression.js\\"
    }
  });
}"
`;

exports[`export-metadata export named works 1`] = `
"/* ExportNamedDeclaration with Specifiers */
let foo = 5;

let bar = () => {};

let baz = 'baz';
export { foo as default, bar as foobar, baz };
/* ExportNamedDeclaration with Variable declarations */

if (typeof baz !== 'undefined' && baz && baz === Object(baz) && Object.isExtensible(baz) && !baz.hasOwnProperty('__filemeta')) {
  Object.defineProperty(baz, '__filemeta', {
    configurable: true,
    value: {
      name: \\"baz\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

if (typeof bar !== 'undefined' && bar && bar === Object(bar) && Object.isExtensible(bar) && !bar.hasOwnProperty('__filemeta')) {
  Object.defineProperty(bar, '__filemeta', {
    configurable: true,
    value: {
      name: \\"bar\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

if (typeof foo !== 'undefined' && foo && foo === Object(foo) && Object.isExtensible(foo) && !foo.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

export let foo1 = 5,
    bar1 = () => {};
/* ExportNamedDeclaration with Function and Class declarations */

if (typeof bar1 !== 'undefined' && bar1 && bar1 === Object(bar1) && Object.isExtensible(bar1) && !bar1.hasOwnProperty('__filemeta')) {
  Object.defineProperty(bar1, '__filemeta', {
    configurable: true,
    value: {
      name: \\"bar1\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

if (typeof foo1 !== 'undefined' && foo1 && foo1 === Object(foo1) && Object.isExtensible(foo1) && !foo1.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo1, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo1\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

export function foo2() {}

if (typeof foo2 !== 'undefined' && foo2 && foo2 === Object(foo2) && Object.isExtensible(foo2) && !foo2.hasOwnProperty('__filemeta')) {
  Object.defineProperty(foo2, '__filemeta', {
    configurable: true,
    value: {
      name: \\"foo2\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}

export class Bar2 {}

if (typeof Bar2 !== 'undefined' && Bar2 && Bar2 === Object(Bar2) && Object.isExtensible(Bar2) && !Bar2.hasOwnProperty('__filemeta')) {
  Object.defineProperty(Bar2, '__filemeta', {
    configurable: true,
    value: {
      name: \\"Bar2\\",
      filename: \\"tests/fixtures/export-named/index.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re export default 1`] = `
"/* Re-export  */
import __DOCZ_DUMMY_EXPORT_DEFAULT from \\"../assets/a\\";
export default __DOCZ_DUMMY_EXPORT_DEFAULT;

if (typeof __DOCZ_DUMMY_EXPORT_DEFAULT !== 'undefined' && __DOCZ_DUMMY_EXPORT_DEFAULT && __DOCZ_DUMMY_EXPORT_DEFAULT === Object(__DOCZ_DUMMY_EXPORT_DEFAULT) && Object.isExtensible(__DOCZ_DUMMY_EXPORT_DEFAULT) && !__DOCZ_DUMMY_EXPORT_DEFAULT.hasOwnProperty('__filemeta')) {
  Object.defineProperty(__DOCZ_DUMMY_EXPORT_DEFAULT, '__filemeta', {
    configurable: true,
    value: {
      name: \\"__DOCZ_DUMMY_EXPORT_DEFAULT\\",
      filename: \\"tests/fixtures/re-export/re-export-default.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re export default to name 1`] = `
"/* Re-export  */
export { default as aDefault } from '../assets/a';

if (typeof aDefault !== 'undefined' && aDefault && aDefault === Object(aDefault) && Object.isExtensible(aDefault) && !aDefault.hasOwnProperty('__filemeta')) {
  Object.defineProperty(aDefault, '__filemeta', {
    configurable: true,
    value: {
      name: \\"aDefault\\",
      filename: \\"tests/fixtures/re-export/re-export-rename1.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re export name 1`] = `
"/* Re-export  */
export { a } from '../assets/a';

if (typeof a !== 'undefined' && a && a === Object(a) && Object.isExtensible(a) && !a.hasOwnProperty('__filemeta')) {
  Object.defineProperty(a, '__filemeta', {
    configurable: true,
    value: {
      name: \\"a\\",
      filename: \\"tests/fixtures/re-export/re-export.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re export name to default 1`] = `
"/* Re-export  */
import a from \\"../assets/a\\";
export default a;

if (typeof a !== 'undefined' && a && a === Object(a) && Object.isExtensible(a) && !a.hasOwnProperty('__filemeta')) {
  Object.defineProperty(a, '__filemeta', {
    configurable: true,
    value: {
      name: \\"a\\",
      filename: \\"tests/fixtures/re-export/re-export-rename2.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re export name to name 1`] = `
"/* Re-export  */
export { a as aa } from '../assets/a';

if (typeof aa !== 'undefined' && aa && aa === Object(aa) && Object.isExtensible(aa) && !aa.hasOwnProperty('__filemeta')) {
  Object.defineProperty(aa, '__filemeta', {
    configurable: true,
    value: {
      name: \\"aa\\",
      filename: \\"tests/fixtures/re-export/re-export-rename3.js\\"
    }
  });
}"
`;

exports[`export-metadata re-export re-export default 1`] = `""`;
